File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/DisplayNamePlatformService.java
package top.likoslupus.cellulosesz.api.player;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;
import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult;
import top.likoslupus.cellulosesz.api.text.RichText;

public interface DisplayNamePlatformService {

    PlatformResult<Void> setDisplayName(CellPlayer player, RichText displayName);

    PlatformResult<Void> refreshPlayerInfo(CellPlayer player);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/DisplayNamePlatformService.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult
import top.likoslupus.cellulosesz.api.text.RichText

public interface DisplayNamePlatformService {

    public fun setDisplayName(
        player: CellPlayer,
        displayName: RichText
    ): PlatformResult<Void>

    public fun refreshPlayerInfo(player: CellPlayer): PlatformResult<Void>

}
